src/load.rs

```diff
@@ -158,13 +158,17 @@ impl Loader {
             }),
             _ => bail!("rspfile and rspfile_content need to be both specified"),
         };
+        let hide_success = lookup("hide_success").as_deref() == Some("1");
```

Comment:
I guess the precedent for boolean flags in ninja is the "restat = 1" flag, where it is just a presence/absence check rather than requiring an explicit "1", so maybe we should follow that here?

https://github.com/ninja-build/ninja/blob/2a34463e6ec38dab909af94740070d546354b16c/src/graph.cc#L515

Reply:
<img width="693" alt="image" src="https://github.com/user-attachments/assets/e266b240-7325-43a6-8d52-3ecbe4210e8a" />

yeah the docs say "if present, ..." for these kinds of flags